packages/elements/src/components/file-picker/file-picker.component.ts

```diff
@@ -0,0 +1,277 @@
+          ({ name, file }, index) => {
+            return html` <li>
+              <a
+                ?data-active="${index === this.fileIndex}"
+                tabindex="${index === this.fileIndex ? 0 : -1}"
+                @focusout="${() => this.#handleFocus()}"
+                @click="${() => this.#closePicker()}"
+                class="${classMap({
+                  'border-primary-500': index === this.fileIndex,
+                })} my-1 flex rounded border-2 border-black bg-black p-1 px-2 text-gray-800 outline-none focus-visible:border-primary-200"
+                href="${this.#getFragment(file)}/${name}"
+              >
+                ${file.result?.name}<span class="mx-2">•</span><span class="text-gray-400">${name}</span>
+              </a>
+            </li>`;
+          },
+        )}
+      </ul>
+    `;
+  }
+
+  #handleFocus() {
+    this.shadowRoot?.querySelector('input')?.focus();
+  }
+
+  #prepareMap() {
+    if (this.rootModel == null) {
+      return;
+    }
+
+    const prepareFiles = <T extends FileUnderTestModel | TestFileModel>(
+      result: MetricsResult<T, Metrics | TestMetrics> | undefined,
+      parentPath: string | null = null,
+      allFilesKey: string,
+    ) => {
+      if (result === undefined) {
+        return;
+      }
+
+      if (result.file !== undefined && result.file !== null && result.name !== allFilesKey) {
+        this.#searchMap.set(parentPath == null ? result.name : `${parentPath}/${result.name}`, result.file);
+      }
+
+      if (result.childResults.length === 0) {
+        return;
+      }
+
+      result.childResults.forEach((child) => {
+        if (parentPath !== allFilesKey && parentPath !== null && result.name !== null) {
+          prepareFiles(child, `${parentPath}/${result.name}`, allFilesKey);
+        } else if ((parentPath === allFilesKey || parentPath === null) && result.name !== allFilesKey) {
+          prepareFiles(child, result.name, allFilesKey);
+        } else {
+          prepareFiles(child, null, allFilesKey);
+        }
+      });
+    };
+
+    prepareFiles(this.rootModel.systemUnderTestMetrics, null, 'All files');
+    prepareFiles(this.rootModel.testMetrics, null, 'All tests');
+  }
+
+  #handleKeyDown(event: KeyboardEvent) {
+    if (event.ctrlKey && event.key === 'k') {
+      this.#togglePicker(event);
+    } else if (!this.openPicker && event.key === '/') {
+      this.#togglePicker(event);
+    } else if (event.key === 'Escape') {
+      this.#closePicker();
+    } else if (event.key === 'ArrowUp') {
+      this.#handleArrowUp();
+    } else if (event.key === 'ArrowDown') {
+      this.#handleArrowDown();
+    }
+
+    if (event.key === 'ArrowUp' || event.key === 'ArrowDown') {
+      setTimeout(() => {
+        this.#scrollActiveLinkInView();
+      });
+    }
+
+    if (event.key === 'Enter') {
+      this.#handleEnter();
+    }
+  }
+
+  #scrollActiveLinkInView() {
+    const activeLink = this.shadowRoot?.querySelector('a[data-active]');
```

Comment:
Is there a particular reason for using `data-active` and not the `active` psuedo-class?

Reply:
the `data-active` marks it as 'active'. I'm not sure how you would do the same with the `::active` psuedo-class.